scripts/compareAPIs.js

```diff
@@ -2,360 +2,501 @@
+${joinedResult}
+\`\`\``;
+    }
+    diffs.push({iname, result: joinedResult, simplifiedName});
+  });
+
+  return {diffs, name};
 }
 
-// Recursively walks a json object and calls a processing function on each node based on its type ["ADD", "REMOVE", "UPDATE"]
-// tracks the path it's taken through the json object and passes that to the processing function
-function walkChanges(changes, process, path = '') {
-  for (let change of changes) {
-    if (process[change.type]) {
-      process[change.type](change, path);
+// mirrors dev/docs/src/types.js for the most part
+// "rendering" our types to a string instead of React components
+function processType(value) {
+  if (!value) {
+    console.trace('UNTYPED', value);
+    return 'UNTYPED';
+  }
+  if (Object.keys(value).length === 0) {
+    return '{}';
+  }
+  if (value.type === 'any') {
+    return 'any';
+  }
+  if (value.type === 'null') {
+    return 'null';
+  }
+  if (value.type === 'undefined') {
+    return 'undefined';
+  }
+  if (value.type === 'void') {
+    return 'void';
+  }
+  if (value.type === 'unknown') {
+    return 'unknown';
+  }
+  if (value.type === 'never') {
+    return 'never';
+  }
+  if (value.type === 'this') {
+    return 'this';
+  }
+  if (value.type === 'symbol') {
+    return 'symbol';
+  }
+  if (value.type === 'identifier') {
+    return value.name;
+  }
+  if (value.type === 'string') {
+    if (value.value) {
+      return `'${value.value}'`;
+    }
+    return 'string';
+  }
+  if (value.type === 'number') {
+    return 'number';
+  }
+  if (value.type === 'boolean') {
+    return 'boolean';
+  }
+  if (value.type === 'union') {
+    return value.elements.map(processType).join(' | ');
+  }
+  if (value.type === 'intersection') {
+    return `(${value.types.map(processType).join(' & ')})`;
+  }
+  if (value.type === 'application') {
+    let name = value.base.name;
+    if (!name) {
+      name = processType(value.base);
+    }
+    return `${name}<${value.typeParameters.map(processType).join(', ')}>`;
+  }
+  if (value.type === 'typeOperator') {
+    return `${value.operator} ${processType(value.value)}`;
+  }
+  if (value.type === 'function') {
+    return `(${value.parameters.map(processType).join(', ')}) => ${value.return ? processType(value.return) : 'void'}`;
+  }
+  if (value.type === 'parameter') {
+    return processType(value.value);
+  }
+  if (value.type === 'link') {
+    let name = value.id.substr(value.id.lastIndexOf(':') + 1);
+    if (dependantOnLinks.has(currentlyProcessing)) {
+      let foo = dependantOnLinks.get(currentlyProcessing);
+      if (!foo.includes(name)) {
+        foo.push(name);
+      }
+    } else {
+      dependantOnLinks.set(currentlyProcessing, [name]);
     }
-    if (change.changes && change.changes.length >= 0) {
-      walkChanges(change.changes, process, `${path}${path.length > 0 ? `.${change.key}` : change.key}`);
+    return name;
+  }
+  // interface still needed if we have it at top level?
```

Comment:
there's a few todo's in this file, they can be added later to make ts-differ better, not something I think must be done to merge this